rasa/core/channels/rasa_chat.py

```diff
@@ -27,35 +28,58 @@ def from_credentials(cls, credentials):
 
     def __init__(self, url):
         self.base_url = url
+        self.jwt_key = None
+        self.jwt_algorithm = None
 
-    async def _check_token(self, token):
-        url = "{}/auth/verify".format(self.base_url)
-        headers = {"Authorization": token}
-        logger.debug("Requesting user information from auth server {}.".format(url))
-
+    async def _fetch_public_key(self) -> None:
+        public_key_url = "{}/version".format(self.base_url)
         async with aiohttp.ClientSession() as session:
             async with session.get(
-                url, headers=headers, timeout=DEFAULT_REQUEST_TIMEOUT
+                public_key_url, timeout=DEFAULT_REQUEST_TIMEOUT
             ) as resp:
-                if resp.status == 200:
-                    return await resp.json()
-                else:
-                    logger.info(
-                        "Failed to check token: {}. "
-                        "Content: {}".format(token, await resp.text())
+                if resp.status != 200:
+                    logger.info("Failed to fetch ")
+                rjs = await resp.json()
+                if "keys" in rjs:
+                    self.jwt_key = rjs["keys"][0]["key"]
+                    self.jwt_algorithm = rjs["keys"][0]["alg"]
+                    logger.debug(
+                        "Fetched JWT public key for algorithm '{}':\n{}"
+                        "".format(self.jwt_algorithm, self.jwt_key)
                     )
-                    return None
+                else:
+                    logger.info("Could not find JWT public key at `/version` endpoint.")
+
+    async def _decode_jwt(self, bearer_token: Text) -> Dict:
+        bearer_token_prefix = "Bearer "
+        authorization_header_value = bearer_token.replace(bearer_token_prefix, "")
+        return jwt.decode(
+            authorization_header_value, self.jwt_key, algorithms=self.jwt_algorithm
+        )
+
+    async def _decode_bearer_token(self, bearer_token: Text) -> Optional[Dict]:
+        if self.jwt_key is None:
+            await self._fetch_public_key()
+
+        # noinspection PyBroadException
+        try:
+            return await self._decode_jwt(bearer_token)
+        except jwt.exceptions.InvalidSignatureError:
+            logger.error("JWT public key invalid, fetching new one.")
+            await self._fetch_public_key()
+            return await self._decode_jwt(bearer_token)
+        except Exception:
+            logger.exception("Failed to decode bearer token.")
```

Comment:
Can you please also log the exception. Maybe just as debug.

Reply:
will be logged automatically with `logger.exception`